uaclient/entitlements/repo.py

```diff
@@ -165,6 +165,26 @@ def application_status(self) -> "Tuple[ApplicationStatus, str]":
             "{} is not configured".format(self.title),
         )
 
+    def _check_apt_url_delta_should_be_applied(self, delta_apt_url):
+        """Check if apt url delta should be applied.
+
+        :param delta_apt_url: string containing the delta apt url
+            to be applied.
+
+        :return: True if apt url should be applied. False otherwise.
+        """
+        if not delta_apt_url:
+            return False
+
+        apt_file = self.repo_list_file_tmpl.format(name=self.name)
+
+        if os.path.exists(apt_file):
+            with open(apt_file, "r") as f:
+                if delta_apt_url in f.read():
+                    return False
```

Comment:
Let's use our utility functions
```suggestion
            return bool(delta_apt_url in util.load_file(apt_file))
```